client-side/src/components/settings/Notifications.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState,useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next'
+import Select from '../../stories/Select/Select.jsx';
+import GenericInput from '../../stories/GenericInput/genericInput.jsx'
+import CONSTANTS from './constantSetting.js'
+import './Notifications.scss';
+
+const Notifications = ({ currentUser , onUpdate}) => {
+  const { EMAIL_FREQUENCY_ENUM, TITLES, LABELS } = CONSTANTS;
+  const { sendNotificationTime:notificationTime, soundVoice:initialSoundVoice,displayIncomeMessages:showIncomeMessages,
+     displayBrowsingTimeLimit:showBrowsingTimeLimit, emailFrequency: initialEmailFrequency,  } = currentUser.preference;
+  const [emailFrequency, setEmailFrequency] = useState(initialEmailFrequency);
+  const [ringtoneFile, setRingtoneFile] = useState(null);
+  const url = process.env.REACT_APP_BASE_URL;
+  const [soundVoice, setSoundVoice] = useState(`${url}/uploads/${initialSoundVoice}`);
+  const [sendNotificationTime, setSendNotificationTime] = useState(notificationTime);
+  const [displayIncomeMessages, setDisplayIncomeMessages] = useState(showIncomeMessages);
+  const [displayBrowsingTimeLimit, setDisplayBrowsingTimeLimit] = useState(showBrowsingTimeLimit);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    onUpdate({
+      emailFrequency,
+      ringtoneFile,
+      sendNotificationTime,
+      displayIncomeMessages,
+      displayBrowsingTimeLimit
+    });
+  }, [emailFrequency, ringtoneFile, sendNotificationTime, displayIncomeMessages, displayBrowsingTimeLimit, onUpdate]);
+
+  const handleFileChange = (e) => {
+    if (e.target.files[0]) {
+      setRingtoneFile(e.target.files[0]);
+      setSoundVoice(URL.createObjectURL(e.target.files[0]));
+    }
+
+  };
+  const handleChangeEmailFreq = (selectedFrequency) => {
+    if (!Object.keys(EMAIL_FREQUENCY_ENUM).includes(selectedFrequency.toUpperCase())) {
```

Comment:
please export !Object.keys(EMAIL_FREQUENCY_ENUM).includes(selectedFrequency.toUpperCase()) to a variable

Reply:
done